handler/consumer/json/src/test/java/io/knotx/fragments/handler/consumer/json/JsonFragmentsHandlerConsumerFactoryTest.java

```diff
@@ -15,9 +15,200 @@
 
+  private static final String EXPECTED_FRAGMENT_TYPE = "snippet";
+  private static final String EXPECTED_HEADER = "x-knotx-debug";
+  private static final String EXPECTED_PARAM = "debug";
+  private static final String PARAM_OPTION = "param";
+  private static final String HTML_TYPE = "html";
+  private static final String FRAGMENT_BODY_JSON = "{\"user\": \"admin\"}";
+  private static final String FRAGMENT_BODY_HTML = "\"<div>body</div>\"";
+  private static final String USER_KEY = "user";
+  private static final String KNOTX_FRAGMENT_KEY = "_knotx_fragment";
+  private static final String FRAGMENT_TYPES = "fragmentTypes";
+  private static final String UNSUPPORTED = "unsupported";
+
+  private static Stream<Arguments> provideFragmentTypeAndBody() {
+    return Stream.of( //fragmentType, fragmentBody, supportedTypes
+        Arguments.of(EXPECTED_FRAGMENT_TYPE, FRAGMENT_BODY_JSON, new JsonArray()),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_HTML, new JsonArray()),
+        Arguments.of(EXPECTED_FRAGMENT_TYPE, FRAGMENT_BODY_HTML, new JsonArray()),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_JSON, new JsonArray()),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_HTML, new JsonArray().add(EXPECTED_FRAGMENT_TYPE)),
+        Arguments.of(EXPECTED_FRAGMENT_TYPE, FRAGMENT_BODY_HTML,
+            new JsonArray().add(EXPECTED_FRAGMENT_TYPE)),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_JSON, new JsonArray().add(EXPECTED_FRAGMENT_TYPE)),
+        Arguments.of(EXPECTED_FRAGMENT_TYPE, FRAGMENT_BODY_JSON, new JsonArray().add(UNSUPPORTED)),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_HTML, new JsonArray().add(UNSUPPORTED)),
+        Arguments.of(EXPECTED_FRAGMENT_TYPE, FRAGMENT_BODY_HTML, new JsonArray().add(UNSUPPORTED)),
+        Arguments.of(HTML_TYPE, FRAGMENT_BODY_JSON, new JsonArray().add(UNSUPPORTED))
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideFragmentTypeAndBody")
+  @DisplayName("Fragment body should not be modified when no supported methods specified in configuration")
+  void fragmentBodyShouldNotBeModifiedWhenInvalidConfigurationProvided(String fragmentType,
+      String fragmentBody, JsonArray supportedTypes) {
+    FragmentEvent original = new FragmentEvent(
+        new Fragment(fragmentType, new JsonObject(), fragmentBody));
+    FragmentEvent copy = new FragmentEvent(original.toJson());
+
+    FragmentExecutionLogConsumer tested = new JsonFragmentsHandlerConsumerFactory()
+        .create(new JsonObject()
+            .put(CONDITION_OPTION, new JsonObject().put(HEADER_OPTION, EXPECTED_HEADER))
+            .put(FRAGMENT_TYPES, supportedTypes));
+    tested.accept(new ClientRequest()
+            .setHeaders(MultiMap.caseInsensitiveMultiMap().add(EXPECTED_HEADER, "true")),
+        ImmutableList.of(FragmentExecutionLog.newInstance(original)));
+
+    assertEquals(copy, original);
+  }
+
+  @Test
+  @DisplayName("Fragment should not be modified when supported fragments do not contain fragment type")
```

Comment:
Is it this case: `Arguments.of(HTML_TYPE, FRAGMENT_BODY_HTML, new JsonArray().add(EXPECTED_FRAGMENT_TYPE))` ?